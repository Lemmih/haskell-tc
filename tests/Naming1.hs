module Naming1 where

fn :: a -> a
fn var_a = sub var_a
  where
    sub :: a -> a
    sub another_a = const var_a another_a

const :: a -> b -> b
const _ b = b
