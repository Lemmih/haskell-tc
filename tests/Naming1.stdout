Bindings:
:: âˆ€ a. a â†’  a
  [4mfn var_a = sub var_a[0m
  [4m  where[0m
  [4m    sub :: a -> a[0m
  [4m    sub another_a = const var_a another_a[0m
:: a
  fn [4mvar_a[0m = sub var_a
:: âˆ€ a2. a2 â†’  a2
      [4msub another_a = const var_a another_a[0m
:: a2
      sub [4manother_a[0m = const var_a another_a
:: âˆ€ a b. a â†’  b â†’  b
  [4mconst _ b = b[0m
:: b
  const _ [4mb[0m = b

Proofs:
coercion: (âˆ€ a. a â†’  a) @ a
  fn var_a = [4msub[0m var_a
coercion: a
  fn var_a = sub [4mvar_a[0m
coercion: (âˆ€ a b. a â†’  b â†’  b) @ a a2
      sub another_a = [4mconst[0m var_a another_a
coercion: a
      sub another_a = const [4mvar_a[0m another_a
coercion: a2
      sub another_a = const var_a [4manother_a[0m
coercion: b
  const _ b = [4mb[0m
